### PR TITLE
Streaming related changes: Buffer reader and Connection pool throttling

### PR DESCRIPTION
**Pull request by sanyamsinghal**

### CODE REVIEW

**Comment by sanyamsinghal** [comment]
Latest Jenkins Test Pipeline results:
1. With Debezium: https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/1034/

File: yb-voyager/src/tgtdb/target_db_interface.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ type TargetDB interface {
 	Init() error
 	Finalize()
 	InitConnPool() error
+	PrepareConnPoolForStreaming()
```
Comment (amit-yb) [inline yb-voyager/src/tgtdb/target_db_interface.go:31]: How about just `PrepareForStreaming()`?

File: yb-voyager/cmd/importData.go [hunk 2]
```diff
@@ -363,6 +363,7 @@ func importData(importFileTasks []*ImportFileTask) {
 		if changeStreamingIsEnabled(importType) {
 			displayImportedRowCountSnapshot(importFileTasks)
 			color.Blue("streaming changes to target DB...")
+			tdb.PrepareConnPoolForStreaming()
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:366]: Move this inside the `streamChanges()`.

File: yb-voyager/cmd/eventQueue.go [hunk 3]
```diff
@@ -117,21 +114,23 @@ func (eqs *EventQueueSegment) Close() error {
...
+	var line, currline []byte
 
-	// Scan() return false in case of error but it is handled below by Err()
-	_ = eqs.scanner.Scan()
-	// scanner.Err() returns nil when EOF error
-	line, err := eqs.scanner.Bytes(), eqs.scanner.Err()
-	if err != nil {
+	for isPrefix && err == nil {
+		currline, isPrefix, err = eqs.reader.ReadLine()
+		line = append(line, currline...)
```
Comment (amit-yb) [inline yb-voyager/cmd/eventQueue.go:123]: If `!isPrefix`, reuse `currLine`:
```
if isPrefix {
  line = append(line, currLine...)
} else {
  line = currLine
}
```

**Pull request merged**